### ISSUE TITLE
Credit memo not saved

### ISSUE THREAD

**Issue by evs-xsarus** [mollie/magento2#181]
In our Magento 2.3.1 (community edition), we have the issue that credit memo's aren't saved due to an error received by Mollie. The weird thing is, though, that the refund IS created in Mollie.

Using Mollie 1.6.3

Here is the original payment request and response:
**Payment Request**
```json
{
  "amount": {
    "currency": "EUR",
    "value": "189.00" 
  },
  "description": "XXXXX",
  "billingAddress": {
    "streetAndNumber": "XXXXX",
    "postalCode": "XXXXX",
    "city": "XXXXX",
    "region": null,
    "country": "NL" 
  },
  "redirectUrl": "XXXXX\\/mollie\\/checkout\\/success\\/?order_id=XXXXX&payment_token=XXXXX&utm_nooverride=1",
  "webhookUrl": "XXXXX\\/mollie\\/checkout\\/webhook\\/?isAjax=1",
  "method": "creditcard",
  "issuer": null,
  "metadata": {
    "order_id": "XXXXX",
    "store_id": "1",
    "payment_token": "XXXXX" 
  },
  "locale": null,
  "shippingAddress": {
    "streetAndNumber": "XXXXX",
    "postalCode": "XXXXX",
    "city": "XXXXX",
    "region": null,
    "country": "NL" 
  }
}
```
**Payment Response**
```json
{
  "resource": "payment",
  "id": "XXXXX",
  "mode": "live",
  "amount": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "settlementAmount": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "amountRefunded": null,
  "amountRemaining": null,
  "description": "XXXXX",
  "method": "creditcard",
  "status": "open",
  "createdAt": "2019-07-19T10:19:50+00:00",
  "paidAt": null,
  "canceledAt": null,
  "expiresAt": "2019-07-19T10:36:50+00:00",
  "failedAt": null,
  "profileId": "XXXXX",
  "sequenceType": "oneoff",
  "redirectUrl": "XXXXX\\/mollie\\/checkout\\/success\\/?order_id=XXXXX&payment_token=XXXXX&utm_nooverride=1",
  "webhookUrl": "XXXXX\\/mollie\\/checkout\\/webhook\\/?isAjax=1",
  "mandateId": null,
  "subscriptionId": null,
  "orderId": null,
  "settlementId": null,
  "locale": null,
  "metadata": {
    "order_id": "XXXXX",
    "store_id": "1",
    "payment_token": "XXXXX" 
  },
  "details": null,
  "_links": {
    "self": {
      "href": "https:\\/\\/api.mollie.com\\/v2\\/payments\\/XXXXX",
      "type": "application\\/hal+json" 
    },
    "checkout": {
      "href": "https:\\/\\/pay.mollie.nl\\/payment\\/start\\/XXXXX",
      "type": "text\\/html" 
    },
    "documentation": {
      "href": "https:\\/\\/docs.mollie.com\\/reference\\/v2\\/payments-api\\/create-payment",
      "type": "text\\/html" 
    }
  },
  "_embedded": null,
  "isCancelable": false
}
````
**Payment webhook**
```json
{
  "resource": "payment",
  "id": "XXXXX",
  "mode": "live",
  "amount": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "settlementAmount": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "amountRefunded": {
    "value": "0.00",
    "currency": "EUR" 
  },
  "amountRemaining": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "description": "XXXXX",
  "method": "creditcard",
  "status": "paid",
  "createdAt": "2019-07-19T10:19:50+00:00",
  "paidAt": "2019-07-19T10:21:39+00:00",
  "canceledAt": null,
  "expiresAt": null,
  "failedAt": null,
  "profileId": "XXXXX",
  "sequenceType": "oneoff",
  "redirectUrl": "XXXXX\\/mollie\\/checkout\\/success\\/?order_id=XXXXX&payment_token=XXXXX&utm_nooverride=1",
  "webhookUrl": "XXXXX\\/mollie\\/checkout\\/webhook\\/?isAjax=1",
  "mandateId": null,
  "subscriptionId": null,
  "orderId": null,
  "settlementId": null,
  "locale": null,
  "metadata": {
    "order_id": "XXXXX",
    "store_id": "1",
    "payment_token": "XXXXX" 
  },
  "details": {
    "cardFingerprint": "XXXXX",
    "cardNumber": "XXXXX",
    "cardHolder": "XXXXX",
    "cardAudience": "consumer",
    "cardLabel": "Mastercard",
    "cardCountryCode": "NL",
    "cardSecurity": "3dsecure",
    "feeRegion": "intra-eu" 
  },
  "_links": {
    "self": {
      "href": "https:\\/\\/api.mollie.com\\/v2\\/payments\\/XXXXX",
      "type": "application\\/hal+json" 
    },
    "documentation": {
      "href": "https:\\/\\/docs.mollie.com\\/reference\\/v2\\/payments-api\\/get-payment",
      "type": "text\\/html" 
    }
  },
  "_embedded": null,
  "isCancelable": null
}
```

Afterwards, a refund was performed:

**Refund**
```json
{
  "resource": "payment",
  "id": "XXXXX",
  "mode": "live",
  "amount": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "settlementAmount": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "amountRefunded": {
    "value": "189.00",
    "currency": "EUR" 
  },
  "amountRemaining": {
    "value": "0.00",
    "currency": "EUR" 
  },
  "description": "XXXXX",
  "method": "creditcard",
  "status": "paid",
  "createdAt": "2019-07-19T10:19:50+00:00",
  "paidAt": "2019-07-19T10:21:39+00:00",
  "canceledAt": null,
  "expiresAt": null,
  "failedAt": null,
  "profileId": "XXXXX",
  "sequenceType": "oneoff",
  "redirectUrl": "XXXXX\\/mollie\\/checkout\\/success\\/?order_id=XXXXX&payment_token=XXXXX&utm_nooverride=1",
  "webhookUrl": "XXXXX\\/mollie\\/checkout\\/webhook\\/?isAjax=1",
  "mandateId": null,
  "subscriptionId": null,
  "orderId": null,
  "settlementId": null,
  "locale": null,
  "metadata": {
    "order_id": "XXXXX",
    "store_id": "1",
    "payment_token": "XXXXX" 
  },
  "details": {
    "cardFingerprint": "XXXXX",
    "cardNumber": "XXXXX",
    "cardHolder": "XXXXX",
    "cardAudience": "consumer",
    "cardLabel": "Mastercard",
    "cardCountryCode": "NL",
    "cardSecurity": "3dsecure",
    "feeRegion": "intra-eu" 
  },
  "_links": {
    "self": {
      "href": "https:\\/\\/api.mollie.com\\/v2\\/payments\\/XXXXX",
      "type": "application\\/hal+json" 
    },
    "refunds": {
      "href": "https:\\/\\/api.mollie.com\\/v2\\/payments\\/XXXXX\\/refunds",
      "type": "application\\/hal+json" 
    },
    "documentation": {
      "href": "https:\\/\\/docs.mollie.com\\/reference\\/v2\\/payments-api\\/get-payment",
      "type": "text\\/html" 
    }
  },
  "_embedded": null,
  "isCancelable": null,
  "countryCode": "NL" 
}
```
But an error was received:

```
Error executing API call (422: Unprocessable Entity): The specified amount cannot be refunded. Field: amount. Documentation: https://docs.mollie.com/guides/handling-errors
```
This is strange, because the amount refunded equals the amount pait.

**Comment by Frank-Magmodules** [comment]
Hi @evs-xsarus , thanks for opening this issue, we will investigate/reproduce this and let you know the outcome on this as soon as possible.

**Comment by Frank-Magmodules** [comment]
Hi @evs-xsarus , thanks again for detailed issue report. The issue has been resolved in the [v1.6.5](https://github.com/mollie/magento2/releases/tag/v1.6.5) release. We are closing this issue now but please feel free to reopen the issue if this still occurs.
